src/ssl.jl

```diff
@@ -534,18 +534,22 @@ end
 function Base.unsafe_read(ssl::SSLStream, buf::Ptr{UInt8}, nbytes::UInt)
     nread = 0
     while nread < nbytes
-        (!isopen(ssl) || eof(ssl)) && throw(EOFError())
+        !isopen(ssl) && throw(EOFError())
```

Comment:
is it worth adding your comment about why we no longer check `eof` here?

```suggestion
        # If open, optimistically call `SSL_read_ex` to try to save an `eof` call;
        # if that returns `SSL_WANT_READ` we will call `eof` anyway afterwards.
        !isopen(ssl) && throw(EOFError())
```